HIP/hip-0000-adding-pause–feature-to-HTS.md

```diff
@@ -0,0 +1,82 @@
+- type: Standards Track
+- category: Application
+- status: Draft
+- created: 2021-08-09
+- discussions-to: 
+- updated: 2021-08-31
+
+## **Abstract**
+
+Stablecoin tokens issued on various protocols usually have a pause function included as part of their smart contract. This allows the token issuer to pause the whole smart contract for contract upgrading, emergencies and other scenarios. This function is currently not available in the Hedera Token Service. Therefore, we propose to establish a standard format for pause feature in HTS. 
+
+## **Motivation**
+
+There are several reasons that would justify the need for a pause feature;
+
+1. Compliance with regulatory frameworks across jurisdictions in the case of regulated token issuance (e.g. stablecoins)
+2. Compliance with law enforcement requests
+3. Parity with functionalities offered by other blockchain protocols in the case of multichain asset issuance.
+
+We propose to establish a standard for HTS where a pause feature can be included for tokens deployed via HTS. This will be opt-in and allows token issuers the flexibility of including or excluding the pause functionality based on the token issuers' requirements. 
+
+## **Specification**
+
+To be Completed.
+
+- Add a `pauseKey` to create token spec. The key can do both pause and unpause a token
+
+    ```protobuf
+    message TokenCreateTransactionBody {
+        string name = 1; // The publicly visible name of the token, limited to a UTF-8 encoding of length <tt>tokens.maxSymbolUtf8Bytes</tt>.
+        string symbol = 2; // The publicly visible token symbol, limited to a UTF-8 encoding of length <tt>tokens.maxTokenNameUtf8Bytes</tt>.
+        uint32 decimals = 3; // For tokens of type FUNGIBLE_COMMON - the number of decimal places a token is divisible by. For tokens of type NON_FUNGIBLE_UNIQUE - value must be 0
+        uint64 initialSupply = 4; // Specifies the initial supply of tokens to be put in circulation. The initial supply is sent to the Treasury Account. The supply is in the lowest denomination possible. In the case for NON_FUNGIBLE_UNIQUE Type the value must be 0
+        AccountID treasury = 5; // The account which will act as a treasury for the token. This account will receive the specified initial supply or the newly minted NFTs in the case for NON_FUNGIBLE_UNIQUE Type
+        Key adminKey = 6; // The key which can perform update/delete operations on the token. If empty, the token can be perceived as immutable (not being able to be updated/deleted)
+        ...
+        ...
+        ...
+        Key fee_schedule_key = 20; // The key which can change the token's custom fee schedule; must sign a TokenFeeScheduleUpdate transaction
+        repeated CustomFee custom_fees = 21; // The custom fees to be assessed during a CryptoTransfer that transfers units of this token
+        Key pauseKey = 22; // [New] The key which can pause the token. If empty, pause is not possible
```

Comment:
This sets up permissions for the pause action, but how does the actual pause and unpause take place?

Reply:
Noted. Would finalize that before the end of next Monday, SGT

Reply:
@steven-sheehy Thanks for your suggesting. We've updated with the specification. Hope that the question is covered in the commit.